ext/node/01_node.js

```diff
@@ -27,62 +31,55 @@ function assert(cond) {
 let initialized = false;
 const nodeGlobals = {};
 const nodeGlobalThis = new Proxy(globalThis, {
-  get(_target, prop, _receiver) {
-    if (prop in nodeGlobals) {
-      return nodeGlobals[prop];
+  get(target, prop) {
+    if (ReflectHas(nodeGlobals, prop)) {
+      return ReflectGet(nodeGlobals, prop);
     } else {
-      return globalThis[prop];
+      return ReflectGet(target, prop);
     }
   },
-  set(_target, prop, value) {
-    if (prop in nodeGlobals) {
-      nodeGlobals[prop] = value;
+  set(target, prop, value) {
+    if (ReflectHas(nodeGlobals, prop)) {
+      return ReflectSet(nodeGlobals, prop, value);
     } else {
-      globalThis[prop] = value;
+      return ReflectSet(target, prop, value);
     }
-    return true;
   },
-  deleteProperty(_target, prop) {
-    let success = false;
-    if (prop in nodeGlobals) {
-      delete nodeGlobals[prop];
-      success = true;
-    }
-    if (prop in globalThis) {
-      delete globalThis[prop];
-      success = true;
-    }
-    return success;
+  has(target, prop) {
+    return ReflectHas(nodeGlobals, prop) || ReflectHas(target, prop);
   },
-  ownKeys(_target) {
-    const globalThisKeys = ReflectOwnKeys(globalThis);
+  deleteProperty(target, prop) {
+    return (
+      ReflectDeleteProperty(nodeGlobals, prop) ||
+      ReflectDeleteProperty(target, prop)
+    );
```

Comment:
This changes the logic. I've committed a test that passes on main, but fails on this PR.